### PR TITLE
Refactor dialogs

### PR DESCRIPTION
**Pull request by ThanatosGit**

### CODE REVIEW

File: randovania/gui/widgets/game_connection_window.py [hunk 1]
```diff
@@ -368,10 +373,11 @@ def add_ui_for_builder(self, builder: ConnectorBuilder):
         self.builders_layout.addWidget(ui.group)
 
     @asyncSlot()
-    async def on_upload_nintendont_action(self, builder: NintendontConnectorBuilder):
+    async def on_upload_nintendont_action(self, builder: NintendontConnectorBuilder) -> None:
         nintendont_file = randovania.get_data_path().joinpath("nintendont", "boot.dol")
         if not nintendont_file.is_file():
-            return await async_dialog.warning(self, "Missing Nintendont", "Unable to find a Nintendont executable.")
+            await async_dialog.warning(self, "Missing Nintendont", "Unable to find a Nintendont executable.")
+            return
```
[the remark this answers is not in the document]
Comment (Miepee) [reply randovania/gui/widgets/game_connection_window.py:380]: it's not. it returns the button that was clicked, and on the warning prompt we usually dont care about what button was clicked as there's only "ok" to choose from.

File: randovania/gui/widgets/game_connection_window.py [hunk 2]
```diff
@@ -393,9 +399,9 @@ async def on_upload_nintendont_action(self, builder: NintendontConnectorBuilder)
         finally:
             box.button(QtWidgets.QMessageBox.StandardButton.Ok).setEnabled(True)
 
-    def open_debug_connector_window(self, builder: DebugConnectorBuilder):
+    def open_debug_connector_window(self, builder: DebugConnectorBuilder) -> None:
         connector = self.game_connection.get_connector_for_builder(builder)
         if connector is not None:
             assert isinstance(connector, DebugRemoteConnector)
-            builder.connector_window = DebugConnectorWindow(connector)
-            builder.connector_window.show()
```
[the remark this answers is not in the document]
Comment (henriquegemignani) [reply randovania/gui/widgets/game_connection_window.py:401]: Yes, keep a reference so it's not lost.

File: randovania/gui/widgets/game_connection_window.py [hunk 3]
```diff
@@ -309,11 +311,14 @@ async def _attempt_join(self, builder: ConnectorBuilder):
             return
 
         session_id = self._check_session_data(layout_uuid)
-        try:
-            await self.network_client.listen_to_session(session_id, True)
-            await self.window_manager.ensure_multiplayer_session_window(self.network_client, session_id, self.options)
-        except error.NotAuthorizedForActionError:
-            await async_dialog.warning(self, "Unauthorized", "You're not a member of this session.")
+        if session_id:
```
Comment (henriquegemignani) [inline randovania/gui/widgets/game_connection_window.py:314]: I feel that not having a session id for a session you're trying to login is a considerable severe error that shouldn't be just ignored. It's kinda of "the button should have been disabled instead" kind of situation, so maybe this should just assert.

File: randovania/games/common/gui/connector_prompt_dialog.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+from typing import Any
+
+from PySide6 import QtWidgets
+
+from randovania.gui.dialog.text_prompt_dialog import TextPromptDialog
+
+
+class ConnectorPromptDialog(TextPromptDialog):
+    def __init__(self, **kwargs: dict[str, Any]):
+        super().__init__(**kwargs)
```
Comment (henriquegemignani) [inline randovania/games/common/gui/connector_prompt_dialog.py:12]: Can you add some tests for this class?

**Review by henriquegemignani — CHANGES_REQUESTED** [review]

**Review by henriquegemignani — CHANGES_REQUESTED** [review]
Oh, since you added type hints please add the file to the list in pyproject.toml so mypy becomes required for it.

**Comment by henriquegemignani** [comment]
> > Oh, since you added type hints please add the file to the list in pyproject.toml so mypy becomes required for it.
> 
> How do I make mypy not complain about the keyword arguments? I changed it to `Any` for now. According to this https://stackoverflow.com/a/63550734/6459327 `Unpack` and a `TypedDict` could be added but if I use a `TypedDict`, I could just directly declare all parameters in the `__init__` params instead.

Declaring all the parameters in the `__init__` is the better API IMO because then it's clear what the possible args are.
You can also just expose only the parameters we're actually using in the code right now too.

**Review by henriquegemignani — APPROVED** [review]

**Pull request merged**